src/Message/ServerRestPurchaseRequest.php

```diff
@@ -0,0 +1,117 @@
+<?php
+
+namespace Omnipay\SagePay\Message;
+
+/**
+ * Sage Pay REST Server Purchase Request
+ */
+class ServerRestPurchaseRequest extends AbstractRestRequest
+{
+    public function getService()
+    {
+        return static::SERVICE_REST_TRANSACTIONS;
+    }
+
+    /**
+     * @return string the transaction type
+     */
+    public function getTxType()
+    {
+        return ucfirst(strtolower(static::TXTYPE_PAYMENT));
+    }
+
+    /**
+     * Add the optional token details to the base data.
+     *
+     * @return array
+     */
+    public function getData()
+    {
+        $data = $this->getBasePurchaseData();
+
+        if ($this->getCardIdentifier() && $this->getMerchantSessionKey()) {
+            $data = $this->getPaymentMethodData($data);
+        }
+
+        return $data;
+    }
+
+    /**
+     * The required fields concerning the purchase
+     *
+     * @return array
+     */
+    protected function getBasePurchaseData()
+    {
+        $card = $this->getCard();
+
+        $data = $this->getBaseData();
+
+        $data['transactionType'] = $this->getTxType();
+        $data['vendorTxCode'] = $this->getTransactionId();
+        $data['description'] = $this->getDescription();
+        $data['amount'] = (int) $this->getAmount();
```

Comment:
Should this not be...
```suggestion
        $data['amount'] = $this->getAmountInteger();
```
`getAmount` returns a formatted string for the given currency, including decimals etc, so any subunits would be lost with your int cast.